Sources/Networking/HTTPClient/HTTPResponse.swift

```diff
@@ -14,48 +14,108 @@
 
 import Foundation
 
-struct HTTPResponse<Body: HTTPResponseBody> {
+/// A type that represents an HTTP response
+protocol HTTPResponseType {
+
+    associatedtype Body: HTTPResponseBody
 
     typealias Result = Swift.Result<Self, NetworkError>
     typealias Headers = [AnyHashable: Any]
 
-    var statusCode: HTTPStatusCode
+    var statusCode: HTTPStatusCode { get }
     /// Because this property is a standard Swift dictionary, its keys are case-sensitive.
     /// To perform a case-insensitive header lookup, use the `value(forHeaderField:)` method instead.
+    var responseHeaders: HTTPClient.ResponseHeaders { get }
+    var body: Body { get }
+    var requestDate: Date? { get }
+
+}
+
+struct HTTPResponse<Body: HTTPResponseBody>: HTTPResponseType {
+
+    var statusCode: HTTPStatusCode
     var responseHeaders: HTTPClient.ResponseHeaders
     var body: Body
     var requestDate: Date?
-    var verificationResult: VerificationResult
 
 }
 
 extension HTTPResponse: CustomStringConvertible {
 
-    var description: String {
-        let body: String = {
-            if let bodyDescription = (self.body as? CustomStringConvertible)?.description {
-                return bodyDescription
-            } else {
-                return "\(type(of: self.body))"
-            }
-        }()
+    fileprivate var bodyDescription: String {
+        if let bodyDescription = (self.body as? CustomStringConvertible)?.description {
+            return bodyDescription
+        } else {
+            return "\(type(of: self.body))"
+        }
+    }
 
+    var description: String {
         return """
         HTTPResponse(
             statusCode: \(self.statusCode.rawValue),
-            body: \(body),
+            body: \(self.bodyDescription),
+            requestDate: \(self.requestDate?.description ?? "<>")
+        )
+        """
+    }
+
+}
+
+// MARK: - VerifiedHTTPResponse
+
+struct VerifiedHTTPResponse<Body: HTTPResponseBody>: HTTPResponseType {
```

Comment:
In Android, I'm still keeping only the `HTTPResponse` object (in android that's `HTTPResult`. In Android, that's created within the `ETagManager`, (which I admit, it's a weird responsibility to have there), but we don't need to default to `NOT_REQUESTED`.

Reply:
I kept this refactor cause it made things cleaner on iOS.